extendeddatasquare_test.go

```diff
@@ -229,3 +230,119 @@ func genRandDS(width int) [][]byte {
 	}
 	return ds
 }
+
+func TestFlattenedEDS(t *testing.T) {
+	example := createExampleEds(t, ShardSize)
+	want := [][]byte{
+		ones, twos, zeros, threes,
+		threes, fours, eights, fifteens,
+		twos, elevens, thirteens, fours,
+		zeros, thirteens, fives, eights,
+	}
+
+	got := example.FlattenedEDS()
+	assert.Equal(t, want, got)
+}
+
+func TestFlattenedODS(t *testing.T) {
+	example := createExampleEds(t, ShardSize)
+	want := [][]byte{
+		ones, twos,
+		threes, fours,
+	}
+
+	got := example.FlattenedsODS()
+	assert.Equal(t, want, got)
+}
+
+func TestEquals(t *testing.T) {
+	t.Run("returns true for two equal EDS", func(t *testing.T) {
+		a := createExampleEds(t, ShardSize)
+		b := createExampleEds(t, ShardSize)
+		assert.True(t, a.Equals(b))
+	})
+	t.Run("returns false for two unequal EDS", func(t *testing.T) {
+		a := createExampleEds(t, ShardSize)
+
+		type testCase struct {
+			name  string
+			other *ExtendedDataSquare
+		}
+
+		unequalOriginalDataWidth := createExampleEds(t, ShardSize)
+		unequalOriginalDataWidth.originalDataWidth = 1
+
+		unequalCodecs := createExampleEds(t, ShardSize)
+		unequalCodecs.codec = newTestCodec()
+
+		unequalChunkSize := createExampleEds(t, ShardSize*2)
+
+		unequalEds, err := ComputeExtendedDataSquare([][]byte{ones}, NewLeoRSCodec(), NewDefaultTree)
+		require.NoError(t, err)
+
+		testCases := []testCase{
+			{
+				name:  "unequal original data width",
+				other: unequalOriginalDataWidth,
+			},
+			{
+				name:  "unequal codecs",
+				other: unequalCodecs,
+			},
+			{
+				name:  "unequal chunkSize",
+				other: unequalChunkSize,
+			},
+			{
+				name:  "unequalEds",
+				other: unequalEds,
+			},
+		}
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				assert.False(t, a.Equals(tc.other))
+				// Question: reflect.DeepEqual matches the behavior of Equals
+				// for all these test cases. Is it sufficient for clients to use
+				// reflect.DeepEqual and remove Equals from the API?
+				assert.False(t, reflect.DeepEqual(a, tc.other))
```

Comment:
@Wondertan given this observation, can we drop `Equals` from the public API of extended data square?

Reply:
It's better to have Equals, as DeepEqual is using reflect which should be avoided where possible